src/storage/engine/mod.rs

```diff
@@ -357,9 +361,12 @@ impl<I: Iterator> Cursor<I> {
     }
 
     pub fn seek(&mut self, key: &Key, statistics: &mut CFStatistics) -> Result<bool> {
+        self.in_bound = true;
+
         assert_ne!(self.scan_mode, ScanMode::Backward);
         if self.max_key.as_ref().map_or(false, |k| k <= key.encoded()) {
             self.iter.validate_key(key)?;
+            self.in_bound = false;
```

Comment:
do we have a test that validate key return error but in_bound is still true, for this case, the valid will be false or true?

Reply:
That's possible and this PR has related tests! In this case, the cursor should be *not valid* because of a real seek (so iter is not invalid, but in_bound is true). However after this operation, the bound will be updated, so next time it will be in_bound == false.